### ISSUE TITLE
Account for cosine losses

### ISSUE THREAD

**Issue by digiefel** [sswelm/PersistentThrust#8]
Persist the dot product between the engine thrust vector and vessel forward vector to account for cosine losses during persistent thrust

**Comment by sswelm** [comment]
I think we have to assume a thrust vector always aligns with a part transform up vector. Then its just a matter of taking the cosine of the part up transform against the vessel.

**Comment by digiefel** [comment]
Yes, should be a safe assumption anyway; never seen an engine (except maybe some VTOL engines that no one is going to use in PT) that didn't have it thrust on its up direction

**Comment by digiefel** [comment]
Should be working, but still trying to properly test if it is.
